src/lib/ContractParser.js

```diff
@@ -160,14 +161,25 @@ export class ContractParser {
     }
     let interfaces
     if (isErc165) interfaces = await this.getInterfacesERC165(contract)
+    else if (this.isEIP1167(txInputData)) interfaces = { EIP1167: true };
     else interfaces = this.getInterfacesByMethods(methods)
     interfaces = Object.keys(interfaces)
       .filter(k => interfaces[k] === true)
       .map(t => contractsInterfaces[t] || t)
     return { methods, interfaces }
   }
 
-  async getInterfacesERC165 (contract) {
+  getEip1167MasterCopy (bytecode) {
+    const implementationAddress = bytecode.replace('363d3d373d3d3d363d73', '').replace('5af43d82803e903d91602b57fd5bf3', '');
+    return implementationAddress;
+  }
+
+  isEIP1167(bytecode) {
+    const re = new RegExp('^363d3d373d3d3d363d73[a-f0-9]{40}5af43d82803e903d91602b57fd5bf3$', 'i');
+    return re.test(remove0x(bytecode));
+  }
+
+  async getInterfacesERC165(contract) {
```

Comment:
Should we refactor this too?

```suggestion
  getEip1167MasterCopy (bytecode) {
    const implementationAddress = bytecode.replace(EIP_1167_PREFIX, '').replace(EIP_1167_SUFFIX, '');
    return implementationAddress;
  }

  isEIP1167(bytecode) {
    const re = new RegExp(`^${EIP_1167_PREFIX}[a-f0-9]{40}${EIP_1167_SUFFIX}$`, 'i');
    return re.test(remove0x(bytecode));
  }

  async getInterfacesERC165(contract) {
```